openmc/weight_windows.py

```diff
@@ -410,3 +406,180 @@ def from_hdf5(cls, group, meshes):
             weight_cutoff=weight_cutoff,
             id=id
         )
+
+
+def wwinp_to_wws(path):
+    """Creates WeightWindows classes from a wwinp file
+
+    Parameters
+    ----------
+    path : str or pathlib.Path object
+        Path to the wwinp file
+
+    Returns
+    -------
+    list of openmc.WeightWindows
+    """
+
+    with open(path) as wwinp:
+        # BLOCK 1
+        header = wwinp.readline().split(None, 4)
+        # read file type, time-dependence, number of
+        # particles, mesh type and problem identifier
+        _if, iv, ni, nr = [int(x) for x in header[:4]]
+        probid = header[4] if len(header) > 4 else ""
+
+        # header value checks
+        if _if != 1:
+            raise ValueError(f'Found incorrect file type, if: {_if}')
+
+        if iv > 1:
+            # read number of time bins for each particle, 'nt(1...ni)'
+            nt = np.fromstring(wwinp.readline(), sep=' ', dtype=int)
+
+            # raise error if time bins are present for now
+            raise ValueError('Time-dependent weight windows '
+                             'are not yet supported')
+        else:
+            nt = ni * [1]
+
+        if nr == 16:
+            raise NotImplementedError('Cylindrical and spherical mesh '
+                                      'types are not yet supported')
+
+        # read number of energy bins for each particle, 'ne(1...ni)'
+        ne = np.fromstring(wwinp.readline(), sep=' ', dtype=int)
+
+        # read coarse mesh dimensions and lower left corner
+        mesh_description = np.fromstring(wwinp.readline(), sep=' ')
+        nfx, nfy, nfz = [int(x) for x in mesh_description[:3]]
+        xyz0 = mesh_description[3:]
+
+        # read cylindrical and spherical mesh vectors if present
+        if nr == 16:
+            # read number of coarse bins
+            line_arr = np.fromstring(wwinp.readline(), sep=' ')
+            ncx, ncy, ncz = [int(x) for x in line_arr[:3]]
+            # read polar vector (x1, y1, z1)
+            xyz1 = line_arr[3:] - xyz0
+            polar_vec = xyz1 / np.linalg.norm(xyz1)
+            # read azimuthal vector (x2, y2, z2)
+            line_arr = np.fromstring(wwinp.readline(), sep=' ')
+            xyz2 = line_arr[:3] - xyz0
+            azimuthal_vec = xyz2 / np.linalg.norm(xyz2)
+            # read geometry type
+            nwg = int(line_arr[-1])
+        elif nr == 10:
+            # read rectilinear data:
+            # number of coarse mesh bins and mesh type
+            ncx, ncy, ncz, nwg = \
+                [int(x) for x in np.fromstring(wwinp.readline(), sep=' ')]
```

Comment:
```suggestion
            ncx, ncy, ncz, nwg = np.fromstring(wwinp.readline(), dtype=int, sep=' ')
```

Reply:
This one won't quite work b/c those values are floats, which `np.fromstring` can't parse correctly with `dtype=int`. Similar to your other suggestions maybe `astype(int)` instead?

Reply:
Went ahead with `np.fromstring(wwinp.readline(), sep=' ').astype(int)` for now.